### PR TITLE
Add debounce utility + test for Navbar 'more' menu behavior

### PR DESCRIPTION
**Pull request by brboi**

### CODE REVIEW

File: addons/wowl/static/src/utils/ts/utility.ts [hunk 1]
```diff
@@ -2,3 +2,31 @@ export function isBrowserChromium(): boolean {
...
+    const args = arguments;
+    function later() {
+      timeout = null;
+      if (!immediate) {
+        func.apply(context, args);
+      }
+    }
+    const callNow = immediate && !timeout;
+    context.env.browser.clearTimeout(timeout);
+    timeout = context.env.browser.setTimeout(later, wait);
```
Comment (brboi) [inline addons/wowl/static/src/utils/ts/utility.ts]: not necessarily called by a component....

File: addons/wowl/static/tests/helpers/utility.ts [hunk 2]
```diff
@@ -42,7 +42,16 @@ export interface TestConfig extends _TestConfig {
...
+  let browser = (config.browser || {}) as OdooConfig["browser"];
+  browser = Object.assign(
+    {
+      debounce: (f: Function) =>
+        function (this: any) {
+          return f.apply(this);
+        },
+    },
+    browser
+  );
```
Comment (brboi) [inline addons/wowl/static/tests/helpers/utility.ts]: default to no implementation + throw an error / warning ?

File: addons/wowl/static/tests/webclient/navbar_tests.ts [hunk 3]
```diff
@@ -105,3 +110,88 @@ QUnit.test("navbar can display systray items ordered based on their sequence", a
...
+    1: { id: 1, children: [10, 11, 12], name: "App0", appID: 1 },
+    10: { id: 10, children: [], name: "Section 10", appID: 1 },
+    11: { id: 11, children: [], name: "Section 11", appID: 1 },
+    12: { id: 12, children: [120, 121, 122], name: "Section 12", appID: 1 },
+    120: { id: 120, children: [], name: "Section 120", appID: 1 },
+    121: { id: 121, children: [], name: "Section 121", appID: 1 },
+    122: { id: 122, children: [], name: "Section 122", appID: 1 },
+  };
+  baseConfig.serverData!.menus = newMenus;
+  const env = await makeTestEnv(baseConfig);
```
Comment (brboi) [inline addons/wowl/static/tests/webclient/navbar_tests.ts:135]:
```suggestion
  const env = await makeTestEnv({...baseConfig, serverData: { ...baseConfig.serverData, menus: newMenus}});
```

File: addons/wowl/static/tests/webclient/navbar_tests.ts [hunk 4]
```diff
@@ -36,7 +37,11 @@ QUnit.module("Navbar", {
       Component: MySystrayItem,
     };
     systray.add(item.name, item);
-    baseConfig = { services, serverData, systray };
+    const browser: Partial<OdooBrowser> = {
+      setTimeout: (handler, delay, args) => (handler as Function).bind(this, args)(),
```
Comment (aab-odoo) [inline addons/wowl/static/tests/webclient/navbar_tests.ts]: Doesn't this work?
`setTimeout: (handler: Functon, delay, args) => handler(args),`

**Review by aab-odoo — APPROVED** [review]

**Pull request merged**